File: app/src/main/java/com/example/bhagavadgita/Retrofit/ApiService.java
package com.example.bhagavadgita.Retrofit;

import com.example.bhagavadgita.Models.Chapters;
import com.example.bhagavadgita.Models.Verses;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiService {

    @GET("chapters/?limit=18")
    Call<List<Chapters>> GetAllChapters();

    @GET("chapters/{chapterNumber}/")
    Call<Chapters> getParticularChapter(@Path("chapterNumber") int chapterNumber);

    @GET("chapters/{chapterNumber}/verses/")
    Call<List<Verses>> getAllVersesFromParticularChapter(@Path("chapterNumber") int chapterNumber);

    @GET("chapters/{chapterNumber}/verses/{versesNumber}/")
    Call<Verses> getParticularVerse(@Path("chapterNumber") int chapterNumber, @Path("versesNumber") int versesNumber);
}

File: app/src/main/java/com/example/bhagavadgita/data/api/ApiService.kt
package com.example.bhagavadgita.data.api

import com.example.bhagavadgita.data.model.Chapters
import com.example.bhagavadgita.data.model.Verses
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET("chapters/?limit=18")
    suspend  fun getAllChapters(): Response<List<Chapters>>

    @GET("chapters/{chapterNumber}/")
    suspend fun getParticularChapter(@Path("chapterNumber") chapterNumber: Int): Response<Chapters>

    @GET("chapters/{chapterNumber}/verses/")
    suspend fun getAllVersesFromParticularChapter(@Path("chapterNumber") chapterNumber: Int): Response<List<Verses>>

    @GET("chapters/{chapterNumber}/verses/{versesNumber}/")
    suspend fun getParticularVerse(
        @Path("chapterNumber") chapterNumber: Int,
        @Path("versesNumber") versesNumber: Int
    ): Response<Verses>

}
